app/app.go

```diff
@@ -97,203 +96,14 @@ func NewApp(
-	)
-
-	app.keeper.staking = *skeeper.SetHooks(
-		staking.NewMultiStakingHooks(
-			app.keeper.distr.Hooks(),
-			app.keeper.slashing.Hooks(),
-		),
-	)
-
-	app.keeper.mint = mint.NewKeeper(
-		cdc,
-		keys[mint.StoreKey],
-		app.keeper.params.Subspace(mint.DefaultParamspace),
-		&skeeper,
-		app.keeper.supply,
-		auth.FeeCollectorName,
-	)
-
-	app.keeper.upgrade = upgrade.NewKeeper(skipUpgradeHeights, keys[upgrade.StoreKey], cdc)
-
-	app.keeper.crisis = crisis.NewKeeper(
-		app.keeper.params.Subspace(crisis.DefaultParamspace),
-		invCheckPeriod,
-		app.keeper.supply,
-		auth.FeeCollectorName,
-	)
-
-	// create evidence keeper with evidence router
-	evidenceKeeper := evidence.NewKeeper(
-		app.cdc, keys[evidence.StoreKey],
-		app.keeper.params.Subspace(evidence.DefaultParamspace),
-		&app.keeper.staking,
-		app.keeper.slashing,
-	)
-	evidenceRouter := evidence.NewRouter()
-
-	// TODO: register evidence routes
-	evidenceKeeper.SetRouter(evidenceRouter)
-
-	app.keeper.evidence = *evidenceKeeper
-
-	// register the proposal types
-	govRouter := gov.NewRouter()
-	govRouter.AddRoute(gov.RouterKey, gov.ProposalHandler).
-		AddRoute(params.RouterKey, params.NewParamChangeProposalHandler(app.keeper.params)).
-		AddRoute(distr.RouterKey, distr.NewCommunityPoolSpendProposalHandler(app.keeper.distr)).
-		AddRoute(upgrade.RouterKey, upgrade.NewSoftwareUpgradeProposalHandler(app.keeper.upgrade))
-
-	app.keeper.gov = gov.NewKeeper(
-		cdc,
-		keys[gov.StoreKey],
-		app.keeper.params.Subspace(gov.DefaultParamspace).WithKeyTable(gov.ParamKeyTable()),
-		app.keeper.supply,
-		&skeeper,
-		govRouter,
-	)
-
-	app.keeper.deployment = deployment.NewKeeper(
-		cdc,
-		keys[deployment.StoreKey],
-	)
-
-	app.keeper.market = market.NewKeeper(
-		cdc,
-		keys[market.StoreKey],
-	)
-
-	app.keeper.provider = provider.NewKeeper(
-		cdc,
-		keys[provider.StoreKey],
-	)
-
-	app.mm = module.NewManager(
-		genutil.NewAppModule(app.keeper.acct, app.keeper.staking, app.BaseApp.DeliverTx),
-		auth.NewAppModule(app.keeper.acct),
-		bank.NewAppModule(app.keeper.bank, app.keeper.acct),
-
-		supply.NewAppModule(app.keeper.supply, app.keeper.acct),
-		distr.NewAppModule(app.keeper.distr, app.keeper.acct, app.keeper.supply, app.keeper.staking),
-
-		mint.NewAppModule(app.keeper.mint),
-		slashing.NewAppModule(app.keeper.slashing, app.keeper.acct, app.keeper.staking),
-
-		staking.NewAppModule(app.keeper.staking, app.keeper.acct, app.keeper.supply),
-
-		gov.NewAppModule(app.keeper.gov, app.keeper.acct, app.keeper.supply),
-		upgrade.NewAppModule(app.keeper.upgrade),
-		evidence.NewAppModule(app.keeper.evidence),
-		crisis.NewAppModule(&app.keeper.crisis),
-
-		// akash
-		deployment.NewAppModule(
-			app.keeper.deployment,
-			app.keeper.market,
-			app.keeper.bank,
-		),
-
-		market.NewAppModule(
-			app.keeper.market,
-			app.keeper.deployment,
-			app.keeper.provider,
-			app.keeper.bank,
-		),
-
-		provider.NewAppModule(app.keeper.provider, app.keeper.bank, app.keeper.market),
-	)
```

Comment:
same here.  for instance, does `ModuleManager` have an `AppendModule()` function or something that can be used in the tag-specific functions?

Reply:
I am not able to find a way to append modules in ModuleManager(). Tried to reduce the amount of duplicate code and left with these.